endpoints/openrtb2/amp_auction.go

```diff
@@ -795,3 +808,22 @@ func setTrace(req *openrtb2.BidRequest, value string) (err error) {
 
 	return nil
 }
+
+// setSeatNonBid populates bidresponse.ext.prebid.seatnonbid if  bidrequest.ext.prebid.returnallbidstatus is true
+func setSeatNonBid(finalExtBidResponse *openrtb_ext.ExtBidResponse, request *openrtb_ext.RequestWrapper, aucResponse *exchange.AuctionResponse) bool {
+	if aucResponse == nil || request == nil {
+		return false
+	}
+	reqExt, err := request.GetRequestExt()
+	if err == nil {
```

Comment:
I think you need to handle the case where `err != nil` as well.

Reply:
Done